plugin/forward/persistent.go

```diff
@@ -132,18 +126,33 @@ func (t *transport) Dial(proto string) (*dns.Conn, bool, error) {
 		proto = "tcp-tls"
 	}
 
-	t.dial <- proto
-	c := <-t.ret
-	return c.c, c.cached, c.err
+	select {
+	case t.dial <- proto:
+	case <-t.stop:
```

Comment:
This way close() will unblock all pending writes to channels, but (I guess) it will not prevent upcoming writes which will hang

Reply:
yes, this is not nice to the caller, but it will force a clean shutdown; so there's that. Should probably double check that this gives servfail back to the caller so it may retry (think it does)